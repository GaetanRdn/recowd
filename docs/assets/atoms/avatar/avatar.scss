@mixin rc-avatar($config) {
  $colors: map-get($config, colors);

  .rc-avatar {
    font-size: 4.8rem;
    width: 4.8rem;
    height: 4.8rem;
    border-radius: 50%;
    border: 2px solid map-get(map-get($colors, primary), border-color);
  }
}
