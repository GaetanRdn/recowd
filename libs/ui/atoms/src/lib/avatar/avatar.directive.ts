import { Directive } from '@angular/core';

@Directive({
  selector: 'rc-material-icon[rcAvatar], img[rcAvatar]',
  standalone: true,
  host: {
    class: 'rc-avatar',
  },
})
export class AvatarDirective {}
