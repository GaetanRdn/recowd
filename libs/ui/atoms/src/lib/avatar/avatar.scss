@mixin rc-avatar {
  .rc-avatar {
    font-size: 4.8rem;
    width: 4.8rem;
    height: 4.8rem;
    border-radius: 50%;
  }
}
