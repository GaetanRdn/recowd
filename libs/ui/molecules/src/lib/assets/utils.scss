@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
