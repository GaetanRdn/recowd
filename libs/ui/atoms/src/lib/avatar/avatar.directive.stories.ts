import { Meta, moduleMetadata, Story } from '@storybook/angular';
import { MaterialIconComponent } from '../material-icon/material-icon.component';
import { BADGE } from '@geometricpanda/storybook-addon-badges';
import { AvatarDirective } from './avatar.directive';

export default {
  title: 'atoms/avatar',
  component: AvatarDirective,
  parameters: {
    badges: [BADGE.STABLE],
    jest: 'avatar.directive',
  },
  decorators: [
    moduleMetadata({
      imports: [AvatarDirective, MaterialIconComponent],
    }),
  ],
} as Meta<AvatarDirective>;

const templateForMaterialIcon: Story<AvatarDirective> = () => ({
  template: `<rc-material-icon rcAvatar>person_pin</rc-material-icon>`,
});

export const WithMaterialIcon = templateForMaterialIcon.bind({});

const templateForCustomImg: Story<AvatarDirective> = () => ({
  template: `<img rcAvatar src="assets/venom.png" alt="avatar"/>`,
});

export const WithCustomImg = templateForCustomImg.bind({});
