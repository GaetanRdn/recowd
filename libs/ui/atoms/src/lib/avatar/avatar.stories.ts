import { Meta, moduleMetadata, Story } from '@storybook/angular';
import { MaterialIconDirective } from '../material-icon/material-icon.directive';
import { BADGE } from '@geometricpanda/storybook-addon-badges';
import { AvatarDirective } from './avatar.directive';

export default {
  title: 'atoms/avatar',
  parameters: {
    badges: [BADGE.STABLE],
    jest: 'avatar.directive',
  },
  decorators: [
    moduleMetadata({
      imports: [AvatarDirective, MaterialIconDirective],
    }),
  ],
} as Meta<AvatarDirective>;

const templateForMaterialIcon: Story<AvatarDirective> = () => ({
  template: `<rc-material-icon rcAvatar>person_pin</rc-material-icon>`,
});

export const WithMaterialIcon = templateForMaterialIcon.bind({});

const templateForCustomImg: Story<AvatarDirective> = () => ({
  template: `<img rcAvatar src="assets/venom.png" alt="avatar"/>`,
});

export const WithCustomImg = templateForCustomImg.bind({});
