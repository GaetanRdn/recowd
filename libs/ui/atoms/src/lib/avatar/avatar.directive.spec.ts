import { Component } from '@angular/core';
import { TestBed, waitForAsync } from '@angular/core/testing';
import { AvatarDirective } from './avatar.directive';
import { TemplateLookup } from '@recowd/test/utils';

describe('AvatarDirective', () => {
  beforeEach(waitForAsync(() =>
    TestBed.configureTestingModule({
      declarations: [
        MaterialIconHostComponent,
        CustomImgHostComponent,
        AvatarDirective,
      ],
    }).compileComponents()));

  it('should create with material icon', () => {
    const templateLookup: TemplateLookup<MaterialIconHostComponent> =
      new TemplateLookup(MaterialIconHostComponent);
    templateLookup.detectChanges();
    expect(templateLookup.firstChildElement).toMatchSnapshot();
  });

  it('should create with custom img', () => {
    const templateLookup: TemplateLookup<CustomImgHostComponent> =
      new TemplateLookup(CustomImgHostComponent);
    templateLookup.detectChanges();
    expect(templateLookup.firstChildElement).toMatchSnapshot();
  });
});

@Component({
  template: ` <rc-material-icon rcAvatar>person_pin</rc-material-icon>`,
})
class MaterialIconHostComponent {}

@Component({
  template: ` <img rcAvatar src="assets/icons/test.svg" alt="avatar" />`,
})
class CustomImgHostComponent {}
