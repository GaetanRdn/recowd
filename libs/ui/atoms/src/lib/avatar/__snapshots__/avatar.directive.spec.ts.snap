// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarDirective should create with custom img 1`] = `
<img
  alt="avatar"
  rcavatar=""
  src="assets/icons/test.svg"
/>
`;

exports[`AvatarDirective should create with material icon 1`] = `
<rc-material-icon
  class="rc-avatar"
  rcavatar=""
>
  person_pin
</rc-material-icon>
`;
